### PR TITLE
Problem: failed tx are ignored in json-rpc apis (backport #502)

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- fix the issue in ethermint by emitting extra events in ante handler.
- add integration test to test it.
- update cosmos-sdk dependency

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Bug Fixes
 
 - [#454](https://github.com/crypto-org-chain/cronos/pull/454) Add back the latest testnet upgrade handler.
+- [cronos#502](https://github.com/crypto-org-chain/cronos/pull/502) Fix failed tx are ignored in json-rpc apis.
```
Comment (yihuang) [inline CHANGELOG.md:8]:
```suggestion
- [cronos#503](https://github.com/crypto-org-chain/cronos/pull/503) Fix failed tx are ignored in json-rpc apis (backport #502.
```

File: go.mod [hunk 2]
```diff
@@ -153,14 +153,14 @@ require (
 replace (
 	// TODO: fix keyring upstream
 	github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-fixes
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.5-0.20220523154235-2921a1c3c918
```
Comment (yihuang) [inline go.mod:156]: https://github.com/cosmos/cosmos-sdk/compare/v0.45.4..2921a1c3c918

File: go.mod [hunk 3]
```diff
@@ -153,14 +153,14 @@ require (
...
 
 	// TODO: remove when middleware will be implemented
 	github.com/cosmos/ibc-go/v2 => github.com/crypto-org-chain/ibc-go/v2 v2.2.0-hooks2
 
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-alpha1-cronos-9
+	github.com/tharsis/ethermint => github.com/yihuang/ethermint v0.10.0-cronos.0.20220524084532-e0be6d185cf1
```
Comment (yihuang) [inline go.mod:163]: https://github.com/yihuang/ethermint/compare/v0.10.0-alpha1-cronos-9...e0be6d185cf1?expand=1

File: go.mod [hunk 4]
```diff
@@ -153,14 +153,14 @@ require (
...
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.5-0.20220523154235-2921a1c3c918
 
 	// TODO: remove when middleware will be implemented
 	github.com/cosmos/ibc-go/v2 => github.com/crypto-org-chain/ibc-go/v2 v2.2.0-hooks2
 
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-alpha1-cronos-9
+	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-alpha1-cronos-10
```
Comment (yihuang) [inline go.mod:163]: https://github.com/crypto-org-chain/ethermint/compare/v0.10.0-alpha1-cronos-9...v0.10.0-alpha1-cronos-10?expand=1

File: go.mod [hunk 5]
```diff
@@ -153,14 +153,14 @@ require (
 replace (
 	// TODO: fix keyring upstream
 	github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-fixes
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.5-0.20220527093140-057a6ca7e74d
```
Comment (yihuang) [inline go.mod:156]: https://github.com/cosmos/cosmos-sdk/compare/v0.45.4...release/v0.45.x

File: go.mod [hunk 6]
```diff
@@ -153,14 +153,14 @@ require (
 replace (
 	// TODO: fix keyring upstream
 	github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-fixes
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.5
```
Comment (yihuang) [inline go.mod:156]: https://github.com/cosmos/cosmos-sdk/compare/v0.45.4...v0.45.5

**Review by thomas-nguy — APPROVED** [review]
I approved the PR but there are a lot of diff between cosmossdk  v0.45.4 and v0.45.5 (especially some errors message on some util functions that may cause undesirable behavior)

Before merge, better test the binary and check that no app hash mismatch can be produce

**Comment by yihuang** [comment]
> I approved the PR but there are a lot of diff between cosmossdk v0.45.4 and v0.45.5 (especially some errors message on some util functions that may cause undesirable behavior)
> 
> Before merge, better test the binary and check that no app hash mismatch can happen

yeah, we'll go through the qa process too when we release the v0.7.1.

**Pull request merged**